### DISCUSSION TITLE
formula issue [Mar 23]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#5393] [Discord Threads]
> 🙋‍♂️ **mamane19**  
#### Anyone know what it might be related to !?  
![image0](https://cdn.discordapp.com/attachments/661905455894888493/1070630623724699719/Screenshot_2023-02-02_at_11.58.45.png)

**Comment by o1lab** [comment]
> 🙋‍♂️ **wkw**  
#### please share your formula, project info and the log  
#  
> 🙋‍♂️ **mamane19**  
#### `SWITCH({currency}, "USD", "USA", "UGX", "Uganda", "NGN", "Nigeria", "KES", "Kenya")`   
####    
#### The above is the formula   
####    
#### Here is the project info:    
#### Node: **v16.17.1**   
#### Arch: **x64**   
#### Platform: **linux**   
#### Docker: **true**   
#### Database: **pg**   
#### ProjectOnRootDB: **false**   
#### RootDB: **pg**   
#### PackageVersion: **0.100.2**  
#  
> 🙋‍♂️ **mamane19**  
#### Even when I try to delete the column it throws the above error: `Cannot read properties of null (reading 'formula')`  
#  
> 🙋‍♂️ **mamane19**  
#### Even when typing the formula in a newly created column there's a read line under the input field that says the same.  
#  
> 🙋‍♂️ **wkw**  
#### when u say "try to delete the column", you mean the currency column or formula column?  
#  
> 🙋‍♂️ **wkw**  
#### if i delete the currency column, the formula will show ERR bc the reference is deleted.  
![image0](https://cdn.discordapp.com/attachments/1070630623846354955/1070901885898334208/image.png)  
#  
> 🙋‍♂️ **wkw**  
#### when i delete formula column, it didn't show the error.  
#  
> 🙋‍♂️ **wkw**  
#### im on 0.104.3 tho  
#  
> 🙋‍♂️ **mamane19**  
#### I meant when trying to delete the formula column. I was able to do the formula a while back but then all of a sudden it stopped working and I can nor delete that column or add a new column using a formula.  
#  
> 🙋‍♂️ **wkw**  
#### any insight from the log?  
#  
> 🙋‍♂️ **wkw**  
#### are u able to create a new formula referencing `currency`?  
#  
> 🙋‍♂️ **mamane19**  
#### Looks like the formula works now. Very strange. Thank you so much!  
#  
> 🙋‍♂️ **mamane19**  
#### Ohh wrong alert. It seems to be happening in one specific view. But on other tables it works fine. That's weird.  
#  
> 🙋‍♂️ **mamane19**  
#### The network tab when I expect says 400, alongside this message: `{"msg":"Cannot read properties of null (reading 'formula')"}`  
#  
> 🙋‍♂️ **wkw**  
#### how about from docker log  
#  
> 🙋‍♂️ **wkw**  
#### currency column is still here right  
#  
> 🙋‍♂️ **mamane19**  
#### ```log   
#### │ S :: TypeError: Cannot read properties of null (reading 'formula')                                                                   │   
#### │     at t.default (/usr/src/app/docker/main.js:1:1229795)                                                                             │   
#### │     at Function.delete (/usr/src/app/docker/main.js:1:48286)                                                                         │   
#### │     at runMicrotasks (<anonymous>)                                                                                                   │   
#### │     at processTicksAndRejections (node:internal/process/task_queues:96:5)                                                            │   
#### │     at async S (/usr/src/app/docker/main.js:1:1440299)                                                                               │   
#### │     at async /usr/src/app/docker/main.js:1:40185                                                                                     │   
#### │ DELETE /api/v1/db/meta/columns/cl_sj6rdb16flevpq 400 60 - 96.074 ms                                                                     
#### ```  
#  
> 🙋‍♂️ **mamane19**  
#### Yes it is still there. Even if I try a simple formula of adding 1 + 1 I get the same error.  
#  
> 🙋‍♂️ **wkw**  
#### have you tried the latest version  
#  
> 🙋‍♂️ **mamane19**  
#### Okay, let me try that and let you know.  
#  
> 🙋‍♂️ **mamane19**  
#### Hey <@495466997304262658> I updated to the latest image but still shows the same error.  
#  
> 🙋‍♂️ **wkw**  
#### are u able to reproduce on a brand new table? if so, pls provide the steps for reproducing the issue  
#  
> 🙋‍♂️ **mamane19**  
#### It works well anywhere else but that table lol. Not sure what's wrong...  
#  
> 🙋‍♂️ **artemblayn**  
#### Hi, can you help me    
#### I have column 1 with a formula in which I have calculated +3 months from column 2 with the date. How do I now calculate -2 weeks from column 2?  
#  
> 🙋‍♂️ **wkw**  
#### u may try DATEADD({column2}, -2, "weeks")  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/1070630623846354955)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
